File: 4-5.py
import cv2 as cv
import sys

from cv2 import IMREAD_COLOR
from cv2 import COLOR_HLS2BGR
from cv2 import COLOR_BGR2GRAY

img_color = cv.imread('ball.jpeg', IMREAD_COLOR)

if img_color is None:
    print('can not read image file')
    sys.exit(1)

img_gray = cv.cvtColor(img_color, COLOR_BGR2GRAY)

img_canny = cv.Canny(img_gray, 90, 180)

img_result = cv.hconcat([img_gray, img_canny])

img_result = cv.vconcat([img_gray, img_canny])

cv.imshow('Result', img_result)
cv.waitKey(0)
cv.destroyAllWindows()

File: 4-5.cpp
#include "iostream" 
#include "opencv2/opencv.hpp"

using namespace std;
using namespace cv;

int main()  {

    Mat img_color = imread("ball.jpeg", IMREAD_COLOR);
    
    if(img_color.empty()){
        cout << "can not read image file" <<endl;
        return -1;
    }
    Mat img_gray;
    cvtColor(img_color, img_gray, COLOR_BGR2GRAY);

    Mat img_canny;
    Canny(img_gray, img_canny, 90, 180);

    vector<Mat> matrices = {img_gray, img_canny};

    Mat img_result;
    hconcat(matrices, img_result);
    // vconcat(matrices, img_result);

    imshow("Result", img_result);

    waitKey(0);
    destroyAllWindows();

    return 0;
}